drivers/sensor/st/stm32_temp/stm32_temp.c

```diff
@@ -121,42 +121,93 @@ static void read_calibration_data(const struct stm32_temp_config *cfg,
 }
 #endif /* HAS_CALIBRATION */
 
-
 static float convert_adc_sample_to_temperature(const struct device *dev)
 {
 	struct stm32_temp_data *data = dev->data;
 	const struct stm32_temp_config *cfg = dev->config;
+	const uint16_t vdda_mv = adc_ref_internal(data->adc);
 	float temperature;
 
-#if defined(HAS_CALIBRATION)
+#if !defined(HAS_CALIBRATION)
+	/**
+	 * Series without calibration (STM32F1/F2):
+	 *   Tjunction = ((Dividend) / Avg_Slope) + 25
+	 *
+	 *  where Dividend is:
+	 *   - (V25 - Vsense) on STM32F1 series ("ntc")
+	 *   - (Vsense - V25) on STM32F2 series
+	 *  and Vsense = (ADC raw data) / ADC_MAX_VALUE * Vdda
+	 *  and ADC_MAX_VALUE = 4095 (12-bit ADC resolution)
+	 *
+	 * References:
+	 *  - RM0008 §11.10 "Temperature sensor" (STM32F100)
+	 *  - RM0041 §10.9  "Temperature sensor" (STM32F101/F102/F103/F105/F107)
+	 *  - RM0033 §10.10 "Temperature sensor" (STM32F2)
+	 */
+	/* Perform multiplication first for higher accuracy */
+	const int vsense = (data->raw * vdda_mv) / 4095;
+
+	if (cfg->is_ntc) {
+		temperature = (float)(cfg->v25 - vsense);
+	} else {
+		temperature = (float)(vsense - cfg->v25);
+	}
+	temperature /= cfg->average_slope;
+	temperature += 25.0f;
+#else /* HAS_CALIBRATION */
 	uint32_t calib[MAX_CALIB_POINTS];
 
 	read_calibration_data(cfg, calib);
 
-	temperature = ((float)data->raw * adc_ref_internal(data->adc)) / cfg->calib_vrefanalog;
-	temperature -= (calib[0] >> cfg->calib_data_shift);
+	const float Sense_Data = ((float)vdda_mv / cfg->calib_vrefanalog) * data->raw;
```

Comment:
Please stick with strict snake_case
```suggestion
	const float sense_data = ((float)vdda_mv / cfg->calib_vrefanalog) * data->raw;
```
Apply everywhere